service-base/src/main/java/org/eclipse/hono/service/resourcelimits/PrometheusBasedResourceLimitChecks.java

```diff
@@ -470,69 +576,61 @@ private Long extractLongValue(final JsonObject response, final Span span) {
     }
 
     /**
-     * Calculates the data volume (in bytes) that is allowed for a tenant for 
-     * the current period based on the mode defined by 
-     * {@link TenantConstants#FIELD_PERIOD_MODE}.
+     * Calculates the effective resource limit for a tenant for the current period from the configured values.
      * <p>
-     * In <em>monthly</em> mode, if the effectiveSince date doesn't fall on the 
-     * first day of the month then the allowed data volume for the tenant is 
-     * calculated as below. For rest of the months and the <em>days</em> mode,
-     * the maxBytes defined by {@link TenantConstants#FIELD_MAX_BYTES} is used directly.
+     * In the <em>monthly</em> mode, if the effectiveSince date doesn't fall on the 
+     * first day of the month then the effective resource limit for the tenant is 
+     * calculated as below.
      * <pre>
-     *             maxBytes 
+     *             configured limit 
      *   ---------------------------------- x No. of days from effectiveSince till lastDay of the targetDateMonth.
      *    No. of days in the current month
      * </pre>
      * <p>
-     * 
-     * @param effectiveSince The point of time on which the data volume limit 
-     *                       came into effect.
-     * @param targetDateTime The target point of time used for the data usage period 
-     *                       calculation.
+     * For rest of the months and the <em>days</em> mode, the configured limit is used directly.
+     *
+     * @param effectiveSince The point of time on which the given resource limit came into effect.
+     * @param currentDateTime The current date and time.
      * @param mode The mode of the period. 
-     * @param maxBytes The maximum allowed bytes defined in configuration by 
-     *                 {@link TenantConstants#FIELD_MAX_BYTES}. 
-     * @return The allowed data-volume in bytes.
+     * @param configuredLimit The configured limit. 
+     * @return The effective resource limit that has been calculated.
      */
-    long calculateDataVolume(
+    long calculateEffectiveLimit(
             final OffsetDateTime effectiveSince,
-            final OffsetDateTime targetDateTime,
+            final OffsetDateTime currentDateTime,
```

Comment:
if this is *always* the current instant then I do not think that you need to pass it in as a parameter, do you?

Reply:
It need not be passed. I will update it.

Reply:
As I checked, there is an existing test namely _verifyEffectiveResourceLimitCalculation_ uses this method. It verifies the calculated limit values under various scenarios. If the current date is always the actual date time then the expected result also keeps changing and unable to verify. It is better if we have this parameter, so that a fixed date time can be passed and the calculated value can be verified.  WDYT?

Reply:
That is also ok and indeed helpful for testing. However, the parameter then should not be named *currentDateTime* but keep its original name FMPOV

Reply:
yes, I will keep the original name.